packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -181,125 +211,560 @@ class CupertinoAlertDialog extends StatelessWidget {
   ///    section when it is long.
   final ScrollController actionScrollController;
 
-  @override
-  Widget build(BuildContext context) {
+  Widget _buildContent() {
     final List<Widget> children = <Widget>[];
 
     if (title != null || content != null) {
-      final Widget titleSection = new _CupertinoAlertTitleSection(
+      final Widget titleSection = new _CupertinoAlertContentSection(
         title: title,
         content: content,
         scrollController: scrollController,
       );
       children.add(new Flexible(flex: 3, child: titleSection));
-      // Add padding between the sections.
-      children.add(const Padding(padding: EdgeInsets.only(top: 8.0)));
     }
 
+    return new Container(
+      color: _kDialogColor,
+      child: new Column(
+        mainAxisSize: MainAxisSize.min,
+        crossAxisAlignment: CrossAxisAlignment.stretch,
+        children: children,
+      ),
+    );
+  }
+
+  Widget _buildActions() {
+    Widget actionSection = new Container(
+      height: 0.0,
+    );
     if (actions.isNotEmpty) {
-      final Widget actionSection = new _CupertinoAlertActionSection(
+      actionSection = new _CupertinoAlertActionSection(
         children: actions,
         scrollController: actionScrollController,
       );
-      children.add(
-        new Flexible(child: actionSection),
-      );
     }
 
-    return new Padding(
-      padding: const EdgeInsets.symmetric(vertical: _kEdgePadding),
-      child: new CupertinoDialog(
-        child: new Column(
-          mainAxisSize: MainAxisSize.min,
-          crossAxisAlignment: CrossAxisAlignment.stretch,
-          children: children,
+    return actionSection;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final double textScaleFactor = MediaQuery.of(context).textScaleFactor;
+    return new _CupertinoAccessibility(
```

Comment:
I don't understand why we need this here? Can't you just use the textScaleFactor directly in the layout builder below?

Reply:
(and pass the flag to _CupertinoDialogRenderWidget directly)

Reply:
There are multiple Widgets that care about whether or not we're in accessibility mode and I didn't want to sprinkle around this convention of "if text scale factor is above X then that means we're in accessibility mode".  With the _CupertinoAccessibility widget that convention is privatized and other widgets merely ask the question they care about "are we in accessibility mode"